src/askui/agent.py

```diff
@@ -340,16 +353,20 @@ def cli(self, command: str) -> None:
         logger.debug("VisionAgent received instruction to execute '%s' on cli", command)
         subprocess.run(command.split(" "))
 
-    def close(self):
+    @telemetry.record_call()
+    def close(self) -> None:
         if self.client:
             self.client.disconnect()
         if self.controller:
             self.controller.stop(True)
+        telemetry.flush()
```

Comment:
What's happend when one of the methods above throws an exception?

Reply:
...then, this won't be executed. Good catch :) 

Should be fixed now :)